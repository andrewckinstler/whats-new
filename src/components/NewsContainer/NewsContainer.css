section {
  display: grid;
  grid-template-columns: 5fr 5fr;
  grid-gap: 20px;
  padding: 20px;
  overflow-y: scroll;
}