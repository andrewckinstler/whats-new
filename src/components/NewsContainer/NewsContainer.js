import React, { Component } from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';


const NewsContainer = ({articles}) => {

  const articleCards = articles.map(({id, headline, img, description, url}) => {
    return (
      <NewsArticle 
        key={id}
        headline={headline}
        img={img}
        description={description}
        url={url}
      />
    )
  })

  return (
    <section className='container'>
      {articleCards}
    </section>
  )
}


export default NewsContainer;