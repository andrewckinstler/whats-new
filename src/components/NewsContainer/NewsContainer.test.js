import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('should create a boilerplate of the container', () => {
    const mockArticles = [
    {
      headline: 'headline',
      img:'img',
      description: 'description',
      url: 'url'
    },
    {
      headline: 'headline1',
      img:'img1',
      description: 'description1',
      url: 'url1'
    },
    {
      headline: 'headline2',
      img:'img2',
      description: 'description2',
      url: 'url2'
    }
    ]

    const wrapper = shallow(<NewsContainer 
      articles={mockArticles}
    />)

    expect(wrapper).toMatchSnapshot()
  })
})