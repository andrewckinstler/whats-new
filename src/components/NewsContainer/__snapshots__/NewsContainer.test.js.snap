// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should create a boilerplate of the container 1`] = `
<section
  className="container"
>
  <NewsArticle
    description="description"
    headline="headline"
    img="img"
    url="url"
  />
  <NewsArticle
    description="description1"
    headline="headline1"
    img="img1"
    url="url1"
  />
  <NewsArticle
    description="description2"
    headline="headline2"
    img="img2"
    url="url2"
  />
</section>
`;
